ogc/src/main/scala/geotrellis/server/ogc/OgcTime.scala

```diff
@@ -0,0 +1,122 @@
+    case (l: OgcTimeInterval, r: OgcTimeInterval) => l |+| r
+    case (l: OgcTimePositions, _: OgcTimeEmpty.type) => l
+    case (l: OgcTimeInterval, _: OgcTimeEmpty.type) => l
+    case (_: OgcTimeEmpty.type, r: OgcTimePositions) => r
+    case (_: OgcTimeEmpty.type, r: OgcTimeInterval) => r
+    case (l, _) => l
+  }
+}
+
+final case class OgcTimePositions(list: NonEmptyList[ZonedDateTime]) extends OgcTime {
+  override def toString: String = list.toList.map(_.toInstant.toString).mkString(", ")
+}
+
+object OgcTimePositions {
+  implicit val timeOrder: Order[ZonedDateTime] = { (l, r) => implicitly[Ordering[ZonedDateTime]].compare(l, r) }
+
+  implicit val ogcTimePositionsSemigroup: Semigroup[OgcTimePositions] = Semigroup.instance { (l, r) =>
+    OgcTimePositions((l.list ::: r.list).distinct)
+  }
+
+  def apply(timePeriod: ZonedDateTime): OgcTimePositions = OgcTimePositions(NonEmptyList(timePeriod, Nil))
+  def apply(timeString: String): OgcTimePositions = apply(ZonedDateTime.parse(timeString))
+  def apply(times: List[ZonedDateTime]): OgcTime =
+    times match {
+      case head :: tail => OgcTimePositions(NonEmptyList(head, tail))
+      case _ => OgcTimeEmpty
+    }
+  def apply(times: List[String])(implicit d: DummyImplicit): OgcTime = apply(times.map(ZonedDateTime.parse))
+}
+
+/**
+  * Represents the TimeInterval and TimePosition types used in TimeSequence requests
+  *
+  * If end is provided, a TimeInterval is assumed. Otherwise, a TimePosition.
+  *
+  * @param start The start time for this TimePosition or TimeInterval
+  * @param end The end time for this TimeInterval. If None, a TimePosition is assumed from the
+  *              start param.
+  * @param interval ISO 8601:2000 provides a syntax for expressing time periods: the designator P,
+  *                 followed by a number of years Y, months M, days D, a time designator T, number
+  *                 of hours H, minutes M, and seconds S. Unneeded elements may be omitted. Here are
+  *                 a few examples:
+  *                 EXAMPLE 1 -  P1Y, 1 year
+  *                 EXAMPLE 2 - P1M10D, 1 month plus 10 days
+  *                 EXAMPLE 3 - PT2H, 2 hours
+  *
+  *                 @note This param is not validated. It is up to the user to ensure that it is
+  *                       encoded directly
+  */
+final case class OgcTimeInterval(start: ZonedDateTime, end: ZonedDateTime, interval: Option[String]) extends OgcTime {
+  override def toString: String =
+    if(start != end) s"${start.toInstant.toString}/${end.toInstant.toString}${interval.map("/" + _).getOrElse("")}"
+    else start.toInstant.toString
+}
+
+object OgcTimeInterval {
+  /**
+   * Merge two OgcTimeInterval instances
+   * This semigroup instance destroys the interval. If you need to retain interval when combining
+   *  instances, perform this operation yourself.
+   */
+  implicit val ogcTimePositionsSemigroup: Semigroup[OgcTimeInterval] = Semigroup.instance { (l, r) =>
```

Comment:
implicit name should be updated. Looks like it was copy pasted from OgcTimePositions above